docs-web-common/src/main/java/com/sismics/util/filter/JwtBasedSecurityFilter.java

```diff
@@ -0,0 +1,131 @@
+import com.sismics.docs.core.dao.UserDao;
+import com.sismics.docs.core.model.jpa.User;
+import com.sismics.feign.KeycloakClient;
+import feign.Feign;
+import feign.gson.GsonDecoder;
+import feign.gson.GsonEncoder;
+import feign.okhttp.OkHttpClient;
+import feign.slf4j.Slf4jLogger;
+import jakarta.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Objects;
+import java.util.UUID;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * This filter is used to authenticate the user having an active session by validating a jwt token.
+ * The filter extracts the jwt token stored from Authorization header.
+ * It validates the token by calling an Identity Broker like KeyCloak.
+ * If validated, the user is retrieved, and the filter injects a UserPrincipal into the request attribute.
+ *
+ * @author smitra
+ */
+public class JwtBasedSecurityFilter extends SecurityFilter {
+    private static final Logger log = LoggerFactory.getLogger(JwtBasedSecurityFilter.class);
+    /**
+     * Name of the cookie used to store the authentication token.
+     */
+    public static final String HEADER_NAME = "Authorization";
+
+    @Override
+    protected User authenticate(final HttpServletRequest request) {
+        log.info("Jwt authentication started");
+        User user = null;
+        String token = extractAuthToken(request).replace("Bearer ", "");
+        DecodedJWT jwt = JWT.decode(token);
+        if (verifyJwt(jwt, token)) {
+            String email = jwt.getClaim("preferred_username").toString();
+            UserDao userDao = new UserDao();
+            user = userDao.getActiveByUsername(email);
+            if (user == null) {
+                user = new User();
+                user.setRoleId(Constants.DEFAULT_USER_ROLE);
+                user.setUsername(email);
+                user.setEmail(email);
+                user.setStorageQuota(10L);
+                user.setPassword(UUID.randomUUID().toString());
+                try {
+                    userDao.create(user, email);
+                    log.info("user created");
+                } catch (Exception e) {
+                    log.info("Error:" + e.getMessage());
+                    return null;
+                }
+            }
+        }
+        return user;
+    }
+
+    private boolean verifyJwt(final DecodedJWT jwt, final String token) {
+
+        try {
+            buildJWTVerifier(jwt).verify(token);
+            // if token is valid no exception will be thrown
+            log.info("Valid TOKEN");
+            return Boolean.TRUE;
+        } catch (CertificateException e) {
+            //if CertificateException comes from buildJWTVerifier()
+            log.info("InValid TOKEN");
+            e.printStackTrace();
```

Comment:
avoid printStackTrace() and add the exception as log.info() argument

Reply:
@jendib have updated all printstacktrace with log.info